quantum/programmable_button.c

```diff
@@ -0,0 +1,51 @@
+/*
+Copyright 2021 Thomas Weißschuh <thomas@t-8ch.de>
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "programmable_button.h"
```

Comment:
Why couldn't these go in `process_programmable_button.c`?

Reply:
Is this comment only for `programmable_button.c` or also for `programmable_button.h`?

The idea was to have the user-facing api in `programmable_button.h` and the internal stuff in `process_programmable_button.h`. Then the respective implementations in the matching `.c` files.

Should I remove the custom public-facing API `.h` or only unify the implementation logic in a single `.c` file?

Reply:
@drashna Could you clarify your last point?